packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-type-form.tsx

```diff
@@ -92,21 +92,22 @@ export const PropertyTypeForm = ({
     setFocus(initialTitle ? "description" : "name");
   }, [initialTitle, setFocus]);
 
-  const { namespace: routeNamespace } = useRouteNamespace();
+  const { activeWorkspaceAccountId, activeWorkspace } =
+    useContext(WorkspaceContext);
 
   const { createPropertyType } = useBlockProtocolCreatePropertyType(
-    routeNamespace?.accountId ?? "",
+    activeWorkspaceAccountId ?? "",
   );
   const { getPropertyType } = useBlockProtocolGetPropertyType();
 
   const generatePropertyTypeBaseUriForUser = (value: string) => {
-    if (!routeNamespace?.shortname) {
+    if (!activeWorkspace || !activeWorkspace.shortname) {
       throw new Error("User shortname must exist");
     }
 
     return generateBaseTypeId({
       domain: frontendUrl,
-      namespace: routeNamespace.shortname,
+      namespace: activeWorkspace.shortname,
       kind: "property-type",
       title: value,
     });
```

Comment:
@nonparibus @CiaranMn @nathggns the implications of this are that if the user creates a property type on an entity type, it's no longer necessarily owned by the same account. It will be owned by whatever is the currently active workspace. This seemed most consistent with what the task outlines, but want to flag this implication nonetheless.

Entity types having property types that are owned by different accounts to my knowledge should not cause any issues on the backend (cc @hashintel/backend).

Reply:
Great, and that is definitely a thing we want to support.

Thanks for flagging this.

We will give thought to the best UX for managing actions (such as creation) across multiple workspaces as we will want to introduce specific switchers -- or at least notices -- throughout the UI (besides the sidebar switcher) eventually. But we want to be consistent in how we do this and have not yet settled on best practice.

Reply:
I agree we want to support this but it seems like a strange default. If I'm editing an entity type in a particular workspace (which I must be in order to create a property type inline and add it to the type), I think it's definitely inintuitive that taking any action in editing the entity type is going to create things in another workspace.

Reply:
To add a property type to an entity type, the user must already have write access to the entity type's owner. Therefore I can easily revert this to default to the workspace the entity type belongs to. Let me know what your thoughts are @nonparibus.

Reply:
Thinking about this, a user is never going to be able to edit an entity type unless they have the workspace it is part of selected in the sidebar.

If they are browsing as any other workspace (including their own user account) they will just see a view page when they visit the entity type.

Reply:
> If they are browsing as any other workspace (including their own user account) they will just see a view page when they visit the entity type.

To be clear that's not currently the case, but if it will be I suppose `activeWorkspace` in this code snippet will always be the entity type's owner's workspace.

Reply:
Can we just make it the entity type owner's ten? 

We might decide that it's annoying to have to switch workspaces to edit things that you own (and we haven't implemented it anyway), but I don't think we will ever want the default in this case to be anything other than the entity type owner's. Just fixing it at that avoids having the odd behavior (a) while we haven't implemented 'must switch workspace to edit' and (b) if we ever decide to revert that / not to introduce it.

Reply:
I think that makes sense, have reverted the changes in https://github.com/hashintel/hash/pull/1555/commits/01bc0e08e1e0b342b00399bf13afb32dad3d1a59